openage/convert/service/export/png/png_tmp_file.pxd

```diff
@@ -0,0 +1,97 @@
+# Copyright 2020-2020 the openage authors. See copying.md for legal info.
+
+from . cimport libpng
+from libc.stdint cimport uint8_t
+
+cdef extern from *:
+    """
+    // Copyright 2021-2021 the openage authors. See copying.md for legal info.
+
+    #include "png.h"
+
+    #include <cstdint>
+    #include <cstdlib>
+    #include <cstring>
+
+    struct tmp_file_buffer_state {
+        char *buffer;
+        size_t size;
```

Comment:
since we are in c++ world, can you use a `std::vector` (or stream, or unique_ptr, ...) here?

Reply:
`libpng` lives in C world still, so keeping it C-like is probably for the best.

Reply:
indeed, but as we need to pass in a memory buffer state when calling this, I'd prefer a c++ variant.

Reply:
This struct only gets used by the callback function passed to libpng. I chose to do it this way bc the code is a lot clearer without having intermediate steps to ensure interoperability between C and C++. This way is likely also more performant, but that's not a big deal since this function isn't getting called every frame or something.

Reply:
@TheJJ `char*` can be directly converted to a Python bytes array by Cython ([see here](https://docs.cython.org/en/latest/src/tutorial/strings.html#passing-byte-strings)), so I stayed with that.